docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java

```diff
@@ -64,5 +89,89 @@ protected ClientYamlTestClient initClientYamlTestClient(ClientYamlSuiteRestSpec
                                                             List<HttpHost> hosts, Version esVersion) throws IOException {
         return new ClientYamlDocsTestClient(restSpec, restClient, hosts, esVersion);
     }
-}
 
+    private static class TestAnalyzer implements ExecutableSection {
+        private static ConstructingObjectParser<TestAnalyzer, XContentLocation> PARSER =
+            new ConstructingObjectParser<>("test_analyzer", false, (a, location) -> {
+                String index = (String) a[0];
+                String builtIn = (String) a[1];
+                String rebuilt = (String) a[2];
+                return new TestAnalyzer(location, index, builtIn, rebuilt);
+            });
+        static {
+            PARSER.declareString(constructorArg(), new ParseField("index"));
+            PARSER.declareString(constructorArg(), new ParseField("built_in"));
+            PARSER.declareString(constructorArg(), new ParseField("rebuilt"));
+        }
+        private static TestAnalyzer parse(XContentParser parser) throws IOException {
+            XContentLocation location = parser.getTokenLocation();
+            TestAnalyzer section = PARSER.parse(parser, location);
+            assert parser.currentToken() == Token.END_OBJECT;
+            parser.nextToken(); // throw out the END_OBJECT to conform with other ExecutableSections
+            return section;
+        }
+
+        private final XContentLocation location;
+        private final String index;
+        private final String builtIn;
+        private final String rebuilt;
+
+        private TestAnalyzer(XContentLocation location, String index, String builtIn, String rebuilt) {
+            this.location = location;
+            this.index = index;
+            this.builtIn = builtIn;
+            this.rebuilt = rebuilt;
+        }
+
+        @Override
+        public XContentLocation getLocation() {
+            return location;
+        }
+
+        @Override
+        public void execute(ClientYamlTestExecutionContext executionContext) throws IOException {
+            int size = 100;
+            List<String> testText = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
+                testText.add(randomRealisticUnicodeOfCodepointLength(between(1, 15))
+                    // Don't look up stashed values
+                    .replace("$", "\\$"));
```

Comment:
Do strings generated from `randomRealisticUnicodeOfCodepointLength` also contain spaces, punctuation marks etc, so that we can test tokenizer part of analyzers?

Reply:
It [doesn't](https://github.com/randomizedtesting/randomizedtesting/blob/master/randomized-runner/src/main/java/com/carrotsearch/randomizedtesting/generators/RealisticUnicodeGenerator.java) look like they do. It'd be cool to insert spaces. I don't think I could easily use the *same* unicode page for both space separated strings though. That might not be too bad though.